bson/src/main/org/bson/codecs/IterableCodec.java

```diff
@@ -79,48 +79,31 @@ public Object transform(final Object objectToTransform) {
 
 
     @Override
-    public Codec<Iterable> withUuidRepresentation(final UuidRepresentation uuidRepresentation) {
```

Comment:
The diffs are a bit ugly due to the mix of functional changes with refactoring.  I didn't realize I would need to extract a superclass until after the functional work was done.  If it's too hard to read, I can go back and rebase everything, doing the refactoring first in a separate PR.